mockrxandroidble/src/main/java/com/polidea/rxandroidble/mockrxandroidble/RxBleClientMock.java

```diff
@@ -271,18 +277,25 @@ public RxBleDevice getBleDevice(@NonNull String macAddress) {
         return createScanOperation(filterServiceUUIDs);
     }
 
+    public void simulateDeviceDiscovery(@NonNull RxBleDeviceMock deviceMock) {
```

Comment:
As we talked. I am also thinking if it would not be good to add method `simulateScanError(@NonNull Throwable throwable)`. But it is a different topic.